Nio/Authentication/LoginView.swift

```diff
@@ -36,6 +37,23 @@ struct LoginContainerView: View {
 
         store.login(username: username, password: password, homeserver: homeserverURL)
     }
+    
+    private func guessHomeserverURL() {
+        if !username.isEmpty && homeserver.isEmpty {
+            let userparts = username.split(separator: ":")
```

Comment:
If we use `components(separatedBy:)` instead we can skip the `String` init in L47.
```suggestion
            let userparts = username.components(separatedBy: ":")
```